Mu2eG4/src/TrackerWireSD.cc

```diff
@@ -30,65 +26,68 @@ namespace mu2e {
-          // I am not sure why we get these cases but we do.  Skip them.
-          if ( (edep == 0. /*|| idep == 0.*/)/*&& step == 0.*/ ) {
-                  if ( _debugList.inList() )  cout<<"Skipped"<<endl;
-                  return false;
-          }
-
-          G4StepPoint* preStepPoint = aStep->GetPreStepPoint();
-          if ( preStepPoint->GetPhysicalVolume()->GetName().find("vol")!=string::npos ) { return false; }
-
-          G4int motherCopyNo = preStepPoint->GetTouchableHandle()->GetReplicaNumber(1);
-
-          if ( _debugList.inList() )  {
-        	  cout<<"Step vol name "<<aStep->GetTrack()->GetVolume()->GetName()<<endl;
-        	  cout<<"Step vol copyNo "<<" mother copyNo "<<motherCopyNo<<endl;
-          }
-
-          // Which process caused this step to end?
-          ProcessCode endCode(_processInfo->
-                              findAndCount(Mu2eG4UserHelpers::findStepStoppingProcessName(aStep)));
-
-          // Add the hit to the framework collection.
-          // The point's coordinates are saved in the mu2e coordinate system.
-          _collection->
-            push_back(StepPointMC(_spHelper->particlePtr(aStep->GetTrack()),
-                                  motherCopyNo,
-                                  aStep->GetTotalEnergyDeposit(),
-                                  aStep->GetNonIonizingEnergyDeposit(),
-                                  aStep->GetPreStepPoint()->GetGlobalTime(),
-                                  aStep->GetPreStepPoint()->GetProperTime(),
-                                  aStep->GetPreStepPoint()->GetPosition() - _mu2eDetCenter,
-                                  aStep->GetPreStepPoint()->GetMomentum(),
-                                  aStep->GetStepLength(),
-                                  endCode
-                                  ));
-
-          return true;
-   }
+    _currentSize += 1;
+
+    if( _sizeLimit>0 && _currentSize>_sizeLimit ) {
+      if( (_currentSize - _sizeLimit)==1 ) {
+        mf::LogWarning("G4") << "Maximum number of particles reached "
+                             << SensitiveDetectorName
+                             << ": "
+                             << _currentSize << endl;
+      }
+      return false;
+    }
+
+    G4double edep  = aStep->GetTotalEnergyDeposit();
+    G4double nidep = aStep->GetNonIonizingEnergyDeposit();
+    G4double step  = aStep->GetStepLength();
+    //G4double idep  = edep-nidep;
+
+    if ( _debugList.inList() )  cout<<"edep "<<edep<<" nidep "<<nidep<<" step "<<step<<endl;
+    // I am not sure why we get these cases but we do.  Skip them.
+    if ( (edep == 0. /*|| idep == 0.*/)/*&& step == 0.*/ ) {
+      if ( _debugList.inList() )  cout<<"Skipped"<<endl;
+      return false;
+    }
+
+    G4StepPoint* preStepPoint = aStep->GetPreStepPoint();
+    if ( preStepPoint->GetPhysicalVolume()->GetName().find("vol")!=string::npos ) {
```

Comment:
I know this is not brought in by the current commit, but why do we have std::string comparisons inside Geant stepping?